mmdet3d/evaluation/metrics/kitti_metric.py

```diff
@@ -74,25 +74,19 @@ def __init__(self,
                 raise KeyError("metric should be one of 'bbox', 'img_bbox', "
                                'but got {metric}.')
 
-    def convert_annos_to_kitti_annos(
-        self,
-        data_annos: list,
-        classes: list = [
-            'Pedestrian', 'Cyclist', 'Car', 'Van', 'Truck', 'Person_sitting',
-            'Tram', 'Misc'
-        ]
-    ) -> list:
+    def convert_annos_to_kitti_annos(self, data_infos: dict) -> list:
         """Convert loading annotations to Kitti annotations.
 
         Args:
-            data_annos (list[dict]): Annotations loaded from ann_file.
-            classes (list[str]): Classes used in the dataset. Default used
-                ['Pedestrian', 'Cyclist', 'Car', 'Van', 'Truck',
-                'Person_sitting', 'Tram', 'Misc'].
+            data_infos (dict): Data infos including metainfo and annotations
+                loaded from ann_file.
 
         Returns:
             List[dict]: List of Kitti annotations.
         """
+        cat2label = data_infos['metainfo']['categories']
```

Comment:
metainfo should not come from data_infos but from the dataset['metainfo']?

Reply:
the dataset['metainfo'] does not include the whole classes information. We need the whole classes info convert the pkl annotations back to Kitti-format annotations